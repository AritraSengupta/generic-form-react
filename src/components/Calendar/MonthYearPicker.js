import React from 'react'
import { Table } from 'semantic-ui-react'

const monthData = [
  ['Jan', 'Feb', 'Mar'],
  ['Apr', 'May', 'Jun'],
  ['Jul', 'Aug', 'Sep'],
  ['Oct', 'Nov', 'Dec']
]

const yearData = (year) => [
  [year - 7, year - 6, year - 5],
  [year - 4, year - 3, year - 2],
  [year - 1, year, year + 1],
  [year + 2, year + 3, year + 4]
]

export class MonthPicker extends React.Component {
  constructor(props) {
    super(props)

    this._clickMonth = this._clickMonth.bind(this)
  }

  processCells = (tableBodyProps) => {
    const { selectedMonth } = this.props
    const cells = []
    tableBodyProps.forEach((month) => {
      const dayObj = {
        key: month,
        content: month || '',
        className: `compact highlight ${
          selectedMonth === month ? 'selected' : ''
        }`,
        onClick: this._clickMonth,
        'data-name': month
      }
      cells.push(dayObj)
    })
    return cells
  }

  renderBodyRowNew = (tableBodyProps, i) => ({
    key: `row-${i}`,
    cells: this.processCells(tableBodyProps)
  })

  _clickMonth(e) {
    const { onSelectMonth } = this.props
    onSelectMonth(e.target.dataset.name)
  }

  render() {
    return (
      <Table
        size='small'
        textAlign='center'
        renderBodyRow={this.renderBodyRowNew}
        tableData={monthData}
        className='table-month'
      />
    )
  }
}

export class YearPicker extends React.Component {
  constructor(props) {
    super(props)

    this._clickYear = this._clickYear.bind(this)
  }

  _clickYear(e) {
    const { onSelectYear } = this.props
    onSelectYear(parseInt(e.target.dataset.name))
  }

  processCells = (tableBodyProps) => {
    const { selectedYear } = this.props
    const cells = []
    tableBodyProps.forEach((year) => {
      const yearObj = {
        key: year,
        content: year || '',
        className: `compact highlight ${
          selectedYear === year ? 'selected' : ''
        }`,
        onClick: this._clickYear,
        'data-name': year
      }
      cells.push(yearObj)
    })
    return cells
  }

  renderBodyRowNew = (tableBodyProps, i) => ({
    key: `row-${i}`,
    cells: this.processCells(tableBodyProps)
  })

  render() {
    const { year } = this.props
    return (
      <Table
        size='small'
        textAlign='center'
        renderBodyRow={this.renderBodyRowNew}
        tableData={yearData(year)}
        className='table-month'
      />
    )
  }
}
