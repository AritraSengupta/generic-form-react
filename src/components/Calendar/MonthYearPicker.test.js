import React from 'react'

import { MonthPicker, YearPicker } from './MonthYearPicker'

describe('Picker', () => {
  it('should render month picker', () => {
    const selectedMonthMock = jest.fn()
    const { container } = render(
      <MonthPicker selectedMonth='Apr' onSelectMonth={selectedMonthMock} />
    )

    expect(container).toMatchSnapshot()
  })

  it('should render year picker', () => {
    const selectYearMock = jest.fn()
    const { container } = render(
      <YearPicker
        year={2019}
        selectedYear={2019}
        onSelectYear={selectYearMock}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
