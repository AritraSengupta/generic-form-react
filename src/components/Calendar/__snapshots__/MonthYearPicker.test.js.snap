// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker should render month picker 1`] = `
<div>
  <table
    class="ui small center aligned table table-month"
  >
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="Jan"
        >
          Jan
        </td>
        <td
          class="compact highlight "
          data-name="Feb"
        >
          Feb
        </td>
        <td
          class="compact highlight "
          data-name="Mar"
        >
          Mar
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="compact highlight selected"
          data-name="Apr"
        >
          Apr
        </td>
        <td
          class="compact highlight "
          data-name="May"
        >
          May
        </td>
        <td
          class="compact highlight "
          data-name="Jun"
        >
          Jun
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="Jul"
        >
          Jul
        </td>
        <td
          class="compact highlight "
          data-name="Aug"
        >
          Aug
        </td>
        <td
          class="compact highlight "
          data-name="Sep"
        >
          Sep
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="Oct"
        >
          Oct
        </td>
        <td
          class="compact highlight "
          data-name="Nov"
        >
          Nov
        </td>
        <td
          class="compact highlight "
          data-name="Dec"
        >
          Dec
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Picker should render year picker 1`] = `
<div>
  <table
    class="ui small center aligned table table-month"
  >
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="2012"
        >
          2012
        </td>
        <td
          class="compact highlight "
          data-name="2013"
        >
          2013
        </td>
        <td
          class="compact highlight "
          data-name="2014"
        >
          2014
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="2015"
        >
          2015
        </td>
        <td
          class="compact highlight "
          data-name="2016"
        >
          2016
        </td>
        <td
          class="compact highlight "
          data-name="2017"
        >
          2017
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="2018"
        >
          2018
        </td>
        <td
          class="compact highlight selected"
          data-name="2019"
        >
          2019
        </td>
        <td
          class="compact highlight "
          data-name="2020"
        >
          2020
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class="compact highlight "
          data-name="2021"
        >
          2021
        </td>
        <td
          class="compact highlight "
          data-name="2022"
        >
          2022
        </td>
        <td
          class="compact highlight "
          data-name="2023"
        >
          2023
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
